TEZ-1037: Replace multiple members in TaskAttemptImpl for container related stuff with a single reference to the container.

ayushtkn opened the pull request.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  25m 58s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  1s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m  7s |  master passed  |
| +1 :green_heart: |  compile  |   0m 45s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   0m 44s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  checkstyle  |   1m 27s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 40s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 28s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +0 :ok: |  spotbugs  |   1m 48s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 46s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 2 new + 79 unchanged - 1 fixed = 81 total (was 80)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  findbugs  |   1m 11s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   5m 13s |  tez-dag in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 11s |  The patch does not generate ASF License warnings.  |
|  |   |  54m 48s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | tez.dag.app.dag.impl.TestTaskImpl |
|   | tez.dag.app.dag.impl.TestDAGRecovery |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/305 |
| JIRA Issue | TEZ-1037 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 21c341e6d214 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 5beab4ced |
| Default Java | Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/1/artifact/out/diff-checkstyle-tez-dag.txt |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/1/artifact/out/patch-unit-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/1/testReport/ |
| Max. process+thread count | 248 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/1/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn commented:
I have 3 checkstyle warnings, 2 for unused import, I will fix it. One for creating getter/setter, which looks overkill to me for test purpose and to make checkstyle happy.
Let me know if that is still a required thing, I can add setter/getter

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  16m 49s |  master passed  |
| +1 :green_heart: |  compile  |   0m 44s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  compile  |   0m 38s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 33s |  master passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m 21s |  master passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +0 :ok: |  spotbugs  |   1m 47s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 45s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 33s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javac  |   0m 33s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 1 new + 79 unchanged - 1 fixed = 80 total (was 80)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09  |
| +1 :green_heart: |  findbugs  |   1m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  6s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  32m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/305 |
| JIRA Issue | TEZ-1037 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 268d13160038 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 5beab4ced |
| Default Java | Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.19+7-post-Ubuntu-0ubuntu122.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~22.04-b09 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/3/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/3/testReport/ |
| Max. process+thread count | 214 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/3/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

abstractdog commented:
I can see that we're doing lot's of interning stuff in the TaskAttemptImpl, which kind of looks like boilerplate from the TaskAttempImpl class' point of view
I guess these intern calls were done for a reason: what about wrapping yarn container into a tez implementation that can hide all the interns + null checks? so
instead of: 
```
StringInterner.intern(RackResolver.resolve(container.getNodeId().getHost()).getNetworkLocation());
```
let it be:
```
container.getNodeRackName();
```
basically, a new Container class can have all these fields that you're about to remove from TaskAttemptImpl

optionally: look for usages of org.apache.hadoop.yarn.api.records.Container in tez code and remove it from there too (it can be done in followup tickets)

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  25m 23s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 34s |  master passed  |
| +1 :green_heart: |  compile  |   0m 41s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  compile  |   0m 37s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   1m 24s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 33s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m 19s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +0 :ok: |  spotbugs  |   1m 47s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 44s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javac  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 1 new + 79 unchanged - 1 fixed = 80 total (was 80)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| -1 :x: |  findbugs  |   1m 19s |  tez-dag generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   5m  4s |  tez-dag in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  50m 55s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:tez-dag |
|  |  org.apache.tez.dag.app.dag.impl.TezContainer defines compareTo(Object) and uses Object.equals()  At TezContainer.java:Object.equals()  At TezContainer.java:[line 34] |
| Failed junit tests | tez.dag.app.dag.impl.TestDAGRecovery |
|   | tez.dag.app.TestSpeculation |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/305 |
| JIRA Issue | TEZ-1037 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 7610d77aa481 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 5beab4ced |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/4/artifact/out/diff-checkstyle-tez-dag.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/4/artifact/out/new-findbugs-tez-dag.html |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/4/artifact/out/patch-unit-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/4/testReport/ |
| Max. process+thread count | 239 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/4/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  25m 57s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 22s |  master passed  |
| +1 :green_heart: |  compile  |   0m 44s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  compile  |   0m 42s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   0m 48s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m 27s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +0 :ok: |  spotbugs  |   1m 29s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 27s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javac  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 1 new + 79 unchanged - 1 fixed = 80 total (was 80)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 11s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| -1 :x: |  findbugs  |   1m 11s |  tez-dag generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   5m 13s |  tez-dag in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   |  50m 48s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:tez-dag |
|  |  org.apache.tez.dag.app.dag.impl.TezContainer defines compareTo(Object) and uses Object.equals()  At TezContainer.java:Object.equals()  At TezContainer.java:[line 34] |
| Failed junit tests | tez.dag.app.dag.impl.TestDAGRecovery |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/305 |
| JIRA Issue | TEZ-1037 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 6470b356afc4 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 5beab4ced |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/5/artifact/out/diff-checkstyle-tez-dag.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/5/artifact/out/new-findbugs-tez-dag.html |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/5/artifact/out/patch-unit-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/5/testReport/ |
| Max. process+thread count | 221 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/5/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 23s |  master passed  |
| +1 :green_heart: |  compile  |   0m 43s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  compile  |   0m 36s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 32s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m 19s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +0 :ok: |  spotbugs  |   1m 45s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 43s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 32s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javac  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  javac  |   0m 27s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 2 new + 79 unchanged - 1 fixed = 81 total (was 80)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| -1 :x: |  findbugs  |   1m 18s |  tez-dag generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   5m  3s |  tez-dag in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 14s |  The patch does not generate ASF License warnings.  |
|  |   |  28m 48s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:tez-dag |
|  |  org.apache.tez.dag.app.dag.impl.TezContainer.equals(Object) is unusual  At TezContainer.java:TezContainer.java:[line 130] |
|  |  org.apache.tez.dag.app.dag.impl.TezContainer defines equals and uses Object.hashCode()  At TezContainer.java:Object.hashCode()  At TezContainer.java:[line 130] |
| Failed junit tests | tez.dag.app.rm.TestTaskScheduler |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/305 |
| JIRA Issue | TEZ-1037 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux db669b6c957d 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 5beab4ced |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/6/artifact/out/diff-checkstyle-tez-dag.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/6/artifact/out/new-findbugs-tez-dag.html |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/6/artifact/out/patch-unit-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/6/testReport/ |
| Max. process+thread count | 201 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/6/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 41s |  master passed  |
| +1 :green_heart: |  compile  |   0m 41s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  compile  |   0m 37s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 32s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m 20s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +0 :ok: |  spotbugs  |   1m 46s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 44s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 32s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javac  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 1 new + 79 unchanged - 1 fixed = 80 total (was 80)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  findbugs  |   1m 15s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  1s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  29m  3s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/305 |
| JIRA Issue | TEZ-1037 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux fcb20ba965e3 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 5beab4ced |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/8/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/8/testReport/ |
| Max. process+thread count | 234 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/8/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn commented:
@abstractdog I have addressed the review comments, added a new TezContainer Class & pushed the methods there, let me know if it looks good.

One checkstyle is left intentionally, that visibility is changed for test & satisfying that looked like an overkill

abstractdog commented:
> @abstractdog I have addressed the review comments, added a new TezContainer Class & pushed the methods there, let me know if it looks good.
> 
> One checkstyle is left intentionally, that visibility is changed for test & satisfying that looked like an overkill

everything looks good, left a minor comment about TezContainer, otherwise +1

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 18s |  master passed  |
| +1 :green_heart: |  compile  |   0m 44s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  compile  |   0m 37s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 32s |  master passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m 20s |  master passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +0 :ok: |  spotbugs  |   1m 47s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 45s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 21s |  tez-dag: The patch generated 2 new + 79 unchanged - 1 fixed = 81 total (was 80)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04  |
| +1 :green_heart: |  javadoc  |   0m  8s |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05  |
| +1 :green_heart: |  findbugs  |   1m 15s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   5m  0s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 14s |  The patch does not generate ASF License warnings.  |
|  |   |  28m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/305 |
| JIRA Issue | TEZ-1037 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 77ba9496ad48 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 5beab4ced |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu122.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~22.04.1-b05 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/9/artifact/out/diff-checkstyle-tez-dag.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/9/testReport/ |
| Max. process+thread count | 217 (vs. ulimit of 5500) |
| modules | C: tez-dag U: tez-dag |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-305/9/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.
